### PR TITLE
[demisto sdk validate] Add BA110 validation

### PR DESCRIPTION
**Pull request by samuelFain**
## Related Issues
fixes:[CIAC-9837](https://jira-dc.paloaltonetworks.com/browse/CIAC-9837)

## Description
Add BA110 validation in new validation format.

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                validator=self,
+                message=self.error_message.format(
+                    self.related_field, content_item.content_type
+                ),
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if any(
+                content_item.content_type.lower() in name.lower()
+                for name in (content_item.name, content_item.display_name)
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py:39]: Correct me if I'm wrong but I think it's safe to test only display_name.

File: demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+class IsEntityTypeInEntityNameValidator(BaseValidator[ContentTypes]):
+    error_code = "BA110"
+    description = (
+        "Check that the entity name or display name does not contain the entity type."
+    )
+    error_message = "The following fields: {0} shouldn't contain the word '{1}'."
+    related_field = "name, display"
+    is_auto_fixable = False
+    related_file_type = [RelatedFileType.YML]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py:25]:
```suggestion
```

File: demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message.format(
+                    self.related_field, content_item.content_type
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py:32]:
```suggestion
                    self.related_field, content_item.content_type
```
needs to be more specific about the field.
I'd try use something like:
'display' if content_item.type == contentType.Integration else 'name'.

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 4]
```diff
@@ -1018,3 +1021,66 @@ def test_IsValidVersionValidator_fix():
...
+            ],
+            "The following fields: name, display shouldn't contain the word 'Playbook'.",
+            id="Playbook Test (valid and invalid)",
+        ),
+    ],
+)
+def test_IsEntityTypeInEntityNameValidator_is_valid(content_items, expected_msg):
+    """
+    Given
+    - Case 1: A content item doest not have its type in name of display felids.
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:1076]: Isn't there 3 tests with 2 content items each?
Let's change to something like 
- 2 integrations
    - One valid integration with ...
    - One invalid integration with...

**Review by YuvHayun — CHANGES_REQUESTED** [review]
See my comments.

File: demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py (hunk 1 above)
Comment (samuelFain) [reply demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py:39]: The `display name` and `name` fields are not necessarily the same (see `ANYRUN` integration for example), so I think its better to check them separately. Leaving the decision to you.

Comment (samuelFain) [reply demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py]: Only Integrations will be tested for both 'name' and 'display' fields, other content items will be tested for 'name' only.

File: demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py (hunk 3 above)
Comment (samuelFain) [reply demisto_sdk/commands/validate/validators/BA_validators/BA110_is_entity_type_in_entity_name.py]: Done.

File: demisto_sdk/commands/validate/tests/BA_validators_test.py (hunk 4 above)
Comment (samuelFain) [reply demisto_sdk/commands/validate/tests/BA_validators_test.py]: Done.

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 5]
```diff
@@ -1018,3 +1021,80 @@ def test_IsValidVersionValidator_fix():
...
+                    paths=["name", "display"],
+                    values=["Test v1", "Testv1"],
+                ),
+                create_integration_object(
+                    paths=["name", "display"],
+                    values=["Test Integration", "TestIntegration"],
+                ),
+            ],
+            "The following fields: name, display shouldn't contain the word 'Integration'.",
+            id="Case 1: Integration Test (valid and invalid)",
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:1041]:
```suggestion
            id="Case 1: Integration in name or display (valid and invalid)",
```

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 6]
```diff
@@ -1018,3 +1021,80 @@ def test_IsValidVersionValidator_fix():
...
+                    paths=["name"],
+                    values=["Test v1"],
+                ),
+                create_script_object(
+                    paths=["name"],
+                    values=["Test Script"],
+                ),
+            ],
+            "The following field: name shouldn't contain the word 'Script'.",
+            id="Case 2: Script Test (valid and invalid)",
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:1055]:
```suggestion
            id="Case 2: Script in name or display (valid and invalid)",
```

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 7]
```diff
@@ -1018,3 +1021,80 @@ def test_IsValidVersionValidator_fix():
...
+                    paths=["name"],
+                    values=["Test v1"],
+                ),
+                create_playbook_object(
+                    paths=["name"],
+                    values=["Test Playbook"],
+                ),
+            ],
+            "The following field: name shouldn't contain the word 'Playbook'.",
+            id="Case 3: Playbook Test (valid and invalid)",
```
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:1069]:
```suggestion
            id="Case 3: Playbook in name or display (valid and invalid)",
```

File: demisto_sdk/commands/validate/tests/BA_validators_test.py
Comment (JasBeilin) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:1]: Lets add the happy case here, only valid files.

**Review by JasBeilin — APPROVED** [review]

**Pull request merged**